### ISSUE TITLE
Figured some things out

### ISSUE THREAD

**Issue by oldmanbluntz** [bepstein111/OctoPrint-Emergency-Stop-FT232H#1]
Hey, working with jneilliii on the discord, some progress has been made. not on this plugin, but it might help you out.

Install OctoPrint on Ubuntu following these instructions:

https://www.illuminated3d.com/2020/06/05/setting-up-octoprint-on-ubuntu-20-04-with-python-3/

At the part where you make the script to start as a service, don't do that yet. go ahead and install adafruit-blinka while still in the venv you already have open from installing octoprint. use these instructions:

https://learn.adafruit.com/circuitpython-on-any-computer-with-ft232h/linux

now, add the environment variable to /etc/environment using the text editor of choice and add "BLINKA_FT232H=1" to the second line.

next step is to create the service but to use systemd/systemctl to create it. Create a file at "/etc/systemd/system/octoprint.service", and put the following code in it, making sure to update the user and the path parts of the file if you didn't install octoprint at the default user "pi".

[Unit]
Description=Octoprint Server
After=network.target
[Service]
Type=simple
Restart=always
RestartSec=10
User=pi
ExecStart=/home/pi/OctoPrint/venv/bin/octoprint serve --port=5000
Environment=BLINKA_FT232H=1
[Install]
WantedBy=multi-user.target

save that, then run

"sudo chmod 777 /etc/systemd/system/octoprint.service"

and finally run

"sudo systemctl enable octoprint"

I have installed a subplugin for psu control written by jneilliii that uses a pin on the FT232H to run an led. The LED is installed on C0 on the FT232H breakout board. I commented out the "os.environ["BLINKA_FT232H"] = "1"" on line 10, not sure if that matters or not. The script is here:

https://gist.githubusercontent.com/jneilliii/d218c2b3e02f603002169686a346b508/raw/93c557c3b0c8a7c9615ce781f5509e5c99e30f7d/psucontrol_subpluginblinka.py

If you install PSUcontrol and that subplugin, it will turn the led on and off.

In your plugin, if you add "adafruit-blinka" to the setup.py, and install octoprint/adafruit-blinka in the way i laid out, and figure out how to get the button press to call the M112 command, it should work for you.

I hope some of this helps you out.

**Comment by oldmanbluntz** [comment]
it was discovered that we needed to set the environment variable to get this to work. i'll grab your plugin and see if i can get it to work, or if any of this helps you out at all if you are still interested in this project.

**Comment by bepstein111** [comment]
I am, but unfortunately have even less time to work on this than I did previously. I suppose I'll eventually be procrastinating something and get to this instead, but it might be a while. I do thank you for the information, I had figured out some of it, but I definitely hadn't figured out the environment variable.

I'm already running two OP servers on one system via systemd services, so adding the line will be easy.
